core/src/main/java/com/linecorp/armeria/internal/SslContextUtil.java

```diff
@@ -65,20 +65,31 @@
 
     public static final List<String> DEFAULT_PROTOCOLS = ImmutableList.of("TLSv1.3", "TLSv1.2");
 
+    private static final List<String> DEFAULT_JDKENGINE_CIPHERS = ImmutableList.<String>builder()
+            .addAll(Http2SecurityUtil.CIPHERS)
+            .build();
+
+    private static final List<String> DEFAULT_JDKENGINE_PROTOCOLS = ImmutableList.of("TLSv1.2");
+
     /**
-     * Creates a {@link SslContext} with Armeria's defaults, enabling support for HTTP/2, TLSv1.3, and
+     * Creates a {@link SslContext} with Armeria's defaults, enabling support for HTTP/2, TLSv1.3 (if supported), and
      * TLSv1.2.
      */
     public static SslContext createSslContext(Supplier<SslContextBuilder> sslContextSupplier,
                                               boolean forceHttp1,
                                               Consumer<? super SslContextBuilder> userCustomizer) {
         final SslContextBuilder builder = sslContextSupplier.get();
 
-        builder.sslProvider(Flags.useOpenSsl() ? SslProvider.OPENSSL : SslProvider.JDK)
-                  .protocols(DEFAULT_PROTOCOLS.toArray(new String[0]));
-
-        builder.ciphers(DEFAULT_CIPHERS, SupportedCipherSuiteFilter.INSTANCE);
-
+        if (Flags.useOpenSsl()) {
+            builder.sslProvider(SslProvider.OPENSSL)
+                   .protocols(DEFAULT_PROTOCOLS.toArray(new String[0]));
+            builder.ciphers(DEFAULT_CIPHERS, SupportedCipherSuiteFilter.INSTANCE);
+        } else {
+            //Netty's JdkSslContext does not support TLSv1.3 by default. In future when it does, we will update.
```

Comment:
Thanks for helping with this! Actually JDK enginre does support it on Java 11+ and we should continue to enable that.

I think you can instead inspect the actual supported protocols using a pattern like this one

https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/common/Flags.java#L286

Reply:
@anuraaga Yes, JDK 11 has TLSv1.3. Looking at https://github.com/netty/netty/blob/ee8206cb2615e1b4e843cf9e52781fdcf2de85f4/handler/src/main/java/io/netty/handler/ssl/JdkSslContext.java#L107, TLSv1.3 is not enabled by default.

Reply:
Should I check - use JdkEngine and running JDK 11, then enable TLSv1.3. Anything else, disable TLSv1.3

Reply:
I don't think you should need to worry about whether or is JDK or 11.

With similar code to the link I gave, you can create a SSLContext, and check the `supportedProtocols`. On Java 8, this should not include TLSv1.3 whole on Java 11 it would.

Reply:
@anuraaga I have updated the code. Can you please check? Testing it on JDK 11, it shows below message. 
`[main] DEBUG io.netty.handler.ssl.JdkSslContext - Default protocols (JDK): [TLSv1.2, TLSv1.1, TLSv1]`